File: dotnet/src/MyDotey.SCF.Simple/Type/String/StringToByteConverter.cs
using System;

namespace MyDotey.SCF.Type.String
{
    /**
    * @author koqizhao
    *
    * May 21, 2018
    */
    public class StringToByteConverter : StringConverter<byte?>
    {
        public static readonly StringToByteConverter Default = new StringToByteConverter();

        public override byte? Convert(string source)
        {
            if (string.IsNullOrWhiteSpace(source))
                return null;

            return byte.Parse(source);
        }
    }
}

File: java/src/main/java/org/mydotey/scf/type/string/StringToByteConverter.java
package org.mydotey.scf.type.string;

/**
 * @author koqizhao
 *
 * May 21, 2018
 */
public class StringToByteConverter extends StringConverter<Byte> {

    public static final StringToByteConverter DEFAULT = new StringToByteConverter();

    public StringToByteConverter() {
        super(Byte.class);
    }

    @Override
    public Byte convert(String source) {
        return Byte.parseByte(source);
    }

}
